### PR TITLE
Updates to inline hooks docs

### PR DESCRIPTION
**Pull request by motikagan-okta**
## Description:
- Updates to existing inline hook documentation, based on feedback.
- Adding information on channel object, used in inline hook management API.
- More information on using the management API and misc edits for consistency.
- Adding short sections on timeouts and security to the overview.
- Reformatting JSON listings.
- Expanding the description of sample use cases for Token Inline Hooks.

### Resolves:

* [OKTA-209417](https://oktainc.atlassian.net/browse/OKTA-209417)

### CODE REVIEW

File: _source/_docs/api/resources/inline-hooks.md [hunk 1]
```diff
@@ -9,30 +9,37 @@ excerpt: The Inline Hooks Management API provides a CRUD interface for registeri
...
-Adds a new Inline Hook to your Organization in `ACTIVE` status.
+Registers a new inline hook to your organization in `ACTIVE` status. You need to pass an [Inline Hook object](#inline-hook-object) in the JSON payload of your request. That object represents the set of required information about the inline hook you are registering, including:
+
+ - The URI of your external service endpoint.
+ - The type of inline hook you are registering.
+ - The secret API key that you want Okta to pass to your external service endpoint (so that your external service can check for its presence as a security measure).
+
+Note that the API key you set here is unrelated to the Okta API token you must supply when making calls to Okta APIs.
+
+You can also ptionally specify extra headers that you wish Okta to pass to your external service with each call.
```
Comment (amoghvasekar-okta) [inline _source/_docs/api/resources/inline-hooks.md]: typo - `o`ptionally

File: _source/_docs/api/resources/inline-hooks.md [hunk 2]
```diff
@@ -299,28 +307,27 @@ curl -v -X PUT \
...
-| Parameter    | Description            | Param Type | DataType | Required |
-|--------------|------------------------|------------|----------|----------|
-| inlineHookId | A valid Inline Hook id | Path       | String   | TRUE     |
+| Parameter | Description                          | Param Type | DataType | Required |
+|-----------|--------------------------------------|------------|----------|----------|
+| `id`      | The ID of the inline hook to delete. | Path       | String   | TRUE     |
 
-Deletes the Inline Hook matching the provided `inlineHook`. Once deleted, this Inline Hook will be unrecoverable.
-As a safety precaution, only Inline Hooks with a status of `INACTIVE` are eligible for deletion.
+Deletes the inline hook matching the provided `id`. Once deleted, the inline hook is unrecoverable. As a safety precaution, only inline hooks with a status of `INACTIVE` are eligible for deletion. To change an inline hook to that status, use an [Update Inline Hook](#update-inline-hook) call, setting the `status` property to `INACTIVE` in the [Inline Hook object](#inline-hook-object).
```
Comment (amoghvasekar-okta) [inline _source/_docs/api/resources/inline-hooks.md]: The recommended way is to use lifecycle methods to update the lifecycle state (`ACTIVE` / `INACTIVE` ) for inline hooks. For example, see - https://developer.okta.com/docs/api/resources/users#lifecycle-operations

We support POST to `/api/v1/inlineHooks/<id>/lifecycle/activate` and `/api/v1/inlineHooks/<id>/lifecycle/deactivate` methods

File: _source/_docs/api/resources/inline-hooks.md [hunk 3]
```diff
@@ -507,49 +510,43 @@ curl -v -X POST \
...
+| id             | Unique key for the Inline Hook.                                                                   | String                            | FALSE    | TRUE   | TRUE     | Assigned                                          |
+| status         | Status of the Inline Hook. `INACTIVE` will block execution.                                       | String                            | FALSE    | FALSE  | FALSE    | Must be either `ACTIVE` or `INACTIVE`.            |
+| name           | Display name for Inline Hook.                                                                     | String                            | FALSE    | TRUE   | FALSE    | Must be between 1 and 255 characters in length.   |
+| type           | Type of the Inline Hook. See list of [Supported Inline Hook Types](#supported-inline-hook-types). | inlineHookType                    | FALSE    | FALSE  | TRUE     | Immutable after Inline Hook creation.             |
+| version        | Version of the Channel.                                                                           | Integer                           | FALSE    | FALSE  | TRUE     | Must match a valid version number.                |
```
Comment (amoghvasekar-okta) [inline _source/_docs/api/resources/inline-hooks.md]: The version is a String, e.g "1.0.0"

File: _source/_docs/api/resources/inline-hooks.md [hunk 4]
```diff
@@ -507,49 +510,43 @@ curl -v -X POST \
...
+| Channel object | Properties of the communications channel used to contact your external service.                   | [Channel](#channel-object) object | FALSE    | FALSE  | FALSE    | Validation is determined by the specific Channel. |
```
Comment (amoghvasekar-okta) [inline _source/_docs/api/resources/inline-hooks.md]: nit : first column should be `channel` for consistency?

File: _source/_docs/api/resources/inline-hooks.md [hunk 5]
```diff
@@ -507,49 +510,43 @@ curl -v -X POST \
...
+| created        | Date of Inline Hook creation.                                                                     | Date                              | TRUE     | FALSE  | TRUE     | Assigned                                          |
+| lastUpdated    | Date of Inline Hook update.                                                                       | Date                              | TRUE     | FALSE  | TRUE     | Assigned                                          |
```
Comment (amoghvasekar-okta) [inline _source/_docs/api/resources/inline-hooks.md]: nit : lowercase `L` for consistency?

File: _source/_use_cases/inline_hooks/index.md [hunk 6]
```diff
@@ -85,6 +85,14 @@ The objects providing this information are nested within a larger object called
 
 Always included is `data.context`, providing context information. In general, `data.context` encapsulates Okta objects that your external service cannot affect, while objects in `data` that are outside of `data.context` encapsulate objects that your external service does have the ability to affect, by means of the commands it sends in its response.
 
+### Timeout and Retry
+
+When Okta calls your external service, it enforces a default timeout of 3 seconds. Okta will attempt at most one retry. A request is not retried if the customer endpoint returns a 4xx HTTP error code. Any 2xx code is considered successful and not retried. If the external service endpoint responds with a redirect, it is not followed.
+
+### Security
+
+To secure the communication channel between Okta and your external service, HTTPS is always used and header-based authentication is implemented in the request.
```
Comment (amoghvasekar-okta) [inline _source/_use_cases/inline_hooks/index.md]: I'd suggest rewording the second part around header-based authentication. Something like `...HTTPS is always used and Okta recommends configuring an AuthScheme to be used in every request to your service`

File: _source/_use_cases/inline_hooks/token_hook/token_hook.md [hunk 7]
```diff
@@ -30,7 +30,7 @@ For steps to enable this inline hook, see below, [Enabling a Token Inline Hook](
 
 This type of inline hook is triggered when OAuth 2.0 and OpenID Connect tokens are minted by your Okta Custom Authorization Server. Before sending the token to the requester, Okta calls out to your external service, and your service can respond with commands to add custom claims to the token.
 
-This functionality can be used to add sensitive user data to tokens without having to store that data in Okta user profiles. For example, tokens minted for a medical app can be augmented with confidential patient data provided by your external service and not stored in Okta.
+This functionality can be used to add data that is sensitive, calculated, or only available in real time, or to add complex data structures not appropriate for inclusion in Okta user profiles. Data added this way is never logged or stored by Okta. As an example, tokens minted for a medical app could be augmented with confidential patient data provided by your external service and not stored in Okta.
```
Comment (caseysoftware) [inline _source/_use_cases/inline_hooks/token_hook/token_hook.md]: Suggestion:
"This functionality can be used to add data that is sensitive, calculated at run time, or complex data structures not appropriate for storing in Okta user profiles."

File: _source/_docs/api/resources/inline-hooks.md (hunk 1 above)
Comment (motikagan-okta) [reply _source/_docs/api/resources/inline-hooks.md]: fixed typo

File: _source/_docs/api/resources/inline-hooks.md (hunk 3 above)
Comment (motikagan-okta) [reply _source/_docs/api/resources/inline-hooks.md]: Changed type to String and mentioned that currently-supported is 1.0.0.

File: _source/_docs/api/resources/inline-hooks.md (hunk 5 above)
Comment (motikagan-okta) [reply _source/_docs/api/resources/inline-hooks.md]: think this might have been fixed before--don't see an upper-case L now.

File: _source/_use_cases/inline_hooks/token_hook/token_hook.md (hunk 7 above)
Comment (motikagan-okta) [reply _source/_use_cases/inline_hooks/token_hook/token_hook.md]: Have changed to: "This functionality can be used to add data that is sensitive, calculated at runtime, or complexly-structured and not appropriate for storing in Okta user profiles. Data added this way is never logged or stored by Okta."

File: _source/_docs/api/resources/inline-hooks.md (hunk 4 above)
Comment (motikagan-okta) [reply _source/_docs/api/resources/inline-hooks.md]: Fixed, thanks for catching.

File: _source/_use_cases/inline_hooks/index.md (hunk 6 above)
Comment (motikagan-okta) [reply _source/_use_cases/inline_hooks/index.md]: Changed to: "To secure the communication channel between Okta and your external service, HTTPS is used for requests, and support is provided for header-based authentication. Okta recommends that you implement an authentication scheme using the authentication header, to be used to authenticate every request received by your external service."

File: _source/_docs/api/resources/inline-hooks.md (hunk 2 above)
Comment (motikagan-okta) [reply _source/_docs/api/resources/inline-hooks.md]: Removed the sentence:
"To change an inline hook to that status, use an [Update Inline Hook](#update-inline-hook) call, setting the `status` property to `INACTIVE` in the [Inline Hook object](#inline-hook-object)."
Will follow up later to add coverage of the lifecycle endpoints.

File: _source/_docs/api/resources/inline-hooks.md [hunk 8]
```diff
@@ -9,30 +9,37 @@ excerpt: The Inline Hooks Management API provides a CRUD interface for registeri
...
+### Create Inline Hook
 
 {% api_operation post /api/v1/inlineHooks%}
 
-Adds a new Inline Hook to your Organization in `ACTIVE` status.
+Registers a new inline hook to your organization in `ACTIVE` status. You need to pass an [Inline Hook object](#inline-hook-object) in the JSON payload of your request. That object represents the set of required information about the inline hook you are registering, including:
+
+ - The URI of your external service endpoint.
+ - The type of inline hook you are registering.
+ - The secret API key that you want Okta to pass to your external service endpoint (so that your external service can check for its presence as a security measure).
```
Comment (amoghvasekar-okta) [inline _source/_docs/api/resources/inline-hooks.md]: nit : this is not required

Comment (motikagan-okta) [reply _source/_docs/api/resources/inline-hooks.md]: Removed API key from list of things that are mandatory to specify when creating an inline hook (but mention afterwards as something that you can specify).
